### ISSUE TITLE
Conditional string formatting in indicator metadata fields

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#1174]
### Addressing a Problem?

Let's take an indicator that accepts `None` for a field:
```
degree_days_exceedance_date = Temp(
    ...
    description="Day of year when the integral of degree days (mean temperature {op} {thresh}) exceeds {sum_thresh}, "
    "with the cumulative sum starting from {after_date}.",
    ...
)
```
The `description` field will return:
```
"Day of year when the integral of degree days (mean temperature > 4 degc) exceeds 200 k days, with the cumulative sum starting from none."
```

### Potential Solution

I haven't delved into the Indicator class code, but if we had access to the call signature keyword arguments, we could perform a string formatting on the fields containing `{...}` patterns.

### Contribution

- [X] I would be willing/able to open a Pull Request to contribute this feature.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aulemahal** [comment]
This is one of the goals of #1120!

Currently you can pass a function to the indicator's definition, it will receive the pre-formatted arguments.
See https://github.com/Ouranosinc/xclim/blob/2d6896a0eb633038446f7a7e9d3f0d5a2c238e8f/xclim/core/indicator.py#L1223-L1224 and lines above to see how `mba` is filled. But this solution is clunky and we almost never use it. And it is not feasible for the translations, while the jinja method could be!

**Comment by Zeitsperre** [comment]
@aulemahal Do your proposed refactoring changes in #2399 open the door to developing this feature?

**Comment by aulemahal** [comment]
I don't think it really changes the implementation potential, but it does point more clearly to which part of the code should be implementing it. That is `xclim.core.indicator._MetadataFormatter`.
